### ISSUE TITLE
add idaapi xref apis

### ISSUE THREAD

**Issue by williballenthin** [williballenthin/python-idb#53]
the analysis module can extract xrefs, but these aren't exposed to the idaapi interface yet. need implementation and tests

**Comment by dukup11ch1** [comment]
i solve it 
```
    def XrefsTo(self,ea,flags=None):
        if flags == idaapi.XREF_ALL:
            typef=[idaapi.fl_JN, idaapi.fl_JF, idaapi.fl_F, idaapi.fl_CN, idaapi.fl_CF]
            typed=[idaapi.dr_U, idaapi.dr_O, idaapi.dr_W, idaapi.dr_R, idaapi.dr_T, idaapi.dr_I]
        if flags == idaapi.XREF_FAR:
            typef=[idaapi.fl_JF, idaapi.fl_F, idaapi.fl_CF]
            typed=None
        if flags == idaapi.XREF_DATA:
            typef=None
            typed=[idaapi.dr_U, idaapi.dr_O, idaapi.dr_W, idaapi.dr_R, idaapi.dr_T, idaapi.dr_I]
        for xref in idb.analysis._get_xrefs(self.idb, dst=ea, tag='X',
                                                types=typef):
            yield xref.src
        for xref in idb.analysis._get_xrefs(self.idb, dst=ea, tag='D',
                                                types=typed):
            yield xref.src

    def XrefsFrom(self,ea,flags=None):
        if flags == idaapi.XREF_ALL:
            typef=[idaapi.fl_JN, idaapi.fl_JF, idaapi.fl_F, idaapi.fl_CN, idaapi.fl_CF]
            typed=[idaapi.dr_U, idaapi.dr_O, idaapi.dr_W, idaapi.dr_R, idaapi.dr_T, idaapi.dr_I]
        if flags == idaapi.XREF_FAR:
            typef=[idaapi.fl_JF, idaapi.fl_F, idaapi.fl_CF]
            typed=None
        if flags == idaapi.XREF_DATA:
            typef=None
            typed=[idaapi.dr_U, idaapi.dr_O, idaapi.dr_W, idaapi.dr_R, idaapi.dr_T, idaapi.dr_I]
        for xref in idb.analysis._get_xrefs(self.idb, dst=ea, tag='X',
                                                types=typef):
            yield xref.dst
        for xref in idb.analysis._get_xrefs(self.idb, dst=ea, tag='D',
                                                types=typed):
            yield xref.dst
```
but i don't know. is it right

**Comment by dukup11ch1** [comment]
this program return xref `Xref = namedtuple('Xref', ['src', 'dst', 'type'])` but 
IDA pro return 'frm' and 'to' instead of 'src' and 'dst'.
![image](https://user-images.githubusercontent.com/34093617/47947916-6a226800-df69-11e8-9690-b62df70827a6.png)


yeah i want to try

**Comment by williballenthin** [comment]
closed in #55
